docs/_docs/components/bar.md

```diff
@@ -165,7 +158,135 @@ Right-aligned vertical bar example:
       </Bar>
     </LayoutSiderContent>
   </LayoutSider>
+  <Layout>
+    <LayoutHeader Fixed="true">
+      Header
+    </LayoutHeader>
+    <LayoutContent>
+      Content
+    </LayoutContent>
+  </Layout>
+</Layout>
+```
+
+### Vertical Bar Toggler
+
+The `BarToggler` for Vertical Bar supports 2 modes:
+- `BarTogglerMode.Normal` (default)
+- `BarTogglerMode.Popout`
+
+These 2 modes, along with an external implementation, provides 3 easy to configure implementations.
+
+**1. Inline**
+
+By simply adding the `BarToggler` anywhere inside the Vertical `Bar` component (at the top level), you will have an inline toggler added to your bar at this location.
+
+Example (used by demo application):
+```html
+<Bar Mode="BarMode.VerticalInline">
+  <BarToggler />
+  <BarBrand>
+    ...
+  </BarBrand>
+  <BarMenu>
+    ...
+  </BarMenu>
+</Bar>
+```
+
+**2. Popout**
+
+Once again, by simply adding the `BarToggler` anywhere inside the Vertical `Bar` component (at the top level), and setting the `Mode` to `Popout` you will have an popout toggler added to your bar at this location.
+
+Example:
+```html
+<Bar Mode="BarMode.VerticalInline">
+  <BarBrand>
+    ...
+  </BarBrand>
+  <BarToggler Mode="BarTogglerMode.Popout" />
+  <BarMenu>
+    ...
+  </BarMenu>
+</Bar>
+```
+
+**3. Extneral** (controlled by Top Bar)
+
+For an externally controlled `BarToggler`, you need to be using the top bar.
+
+Where you have a top bar and vetical bar, you can add multiple `BarTogglers` to the top bar and choose to have one of these control the vertical bar by simply setting the `Bar` property.
+
+In the example below, we will create two `BarTogglers` inside the top bar.
+- One to control the Vertical Bar (closest to it)
+- One to control the Top Bar (at the opposite end)
+
+Example:
+```html
+<Layout Sider="true">
+  <LayoutSider>
+    <LayoutSiderContent>
+      <Bar @ref="sidebar" Mode="BarMode.VerticalInline">
+        ...
+      </Bar>
+    </LayoutSiderContent>
+  </LayoutSider>
+  <Layout>
+    <LayoutHeader Fixed="true">
+      <Bar Mode="BarMode.Horizontal">
+        <BarToggler Bar="sidebar" />
```

Comment:
I like this approach :)

Reply:
PS. Try to have `@` whenever there is variable used. eg. `<BarToggler Bar="@sidebar" />`. I found it a good convention.